src/v/config/property.h

```diff
@@ -702,6 +710,31 @@ bool property<T>::is_array() const {
     return detail::is_array<T>();
 }
 
+template<typename T>
+bool property<T>::is_signed() const {
+    return std::is_signed_v<T>;
+}
+
+template<typename T>
+bool property<T>::is_unsigned() const {
+    return std::is_unsigned_v<T>;
+}
+
+template<typename T>
+bool property<T>::is_bool() const {
+    return std::is_same_v<T, bool>;
+}
```

Comment:
question: do these properly handle configurations that are, for example `std::optional<size_t>`?  Will `is_unsigned()` return true for `retention_bytes` which is such a type?